### ISSUE TITLE
MobileNet v2 vs MobileNet v1 - bug with retrain.py and label_image.py examples from Tensorflow for Poets v2 transfer learning example!

### ISSUE THREAD

**Issue by jgberg** [tensorflow/tensorflow#19071]
### System information
We have reproduced this on Mac OSX, a windows box running a Linux VM, and on a Canadian research cluster all with the same result.
 - mac was v1.7 recompiled to take advantage of CPU: ('v1.7.0-1321-gd82b2f71b6', '1.7.0')
 - stock tensorflow code and examples

### Description
We are using Tensorflow v1.7.  We train **MobileNet v2** (from tfHub - feature/1 version) by running the code Tensorflow provided in **retrain.py** (from examples). When training is done the validation set is run (for us about 700 images) and it reports say **80% validation accuracy**.   Then it saves the model.  We load the saved model using **label_image.py** (also from tensorflow examples) and see how it's doing on a small validation set that we've withheld.  Instead of seeing the validation of 80% roughly confirmed, we instead see a validation accuracy of about 35%.  The clincher is that when use the same code and data and instead use **MobileNet v1** we do see that training validation is confirmed by validation reported using label_image.py . 

When calling label_image we change the input_layer from "input" for MobileNet v1 to "Placeholder" for MobileNet v2

Can you confirm this? What is going on?

### Logs:
From retrain.py:
...
INFO:tensorflow:Initialize variable module/MobilenetV2/expanded_conv_9/project/weights:0 from checkpoint /var/folders/x8/f10sgc052q75r1thg5hlw4dm0000gn/T/tfhub_modules/33f8428fe83945b8b3d46d79168a0e2818e65e8a/variables/variables with MobilenetV2/expanded_conv_9/project/weights
INFO:tensorflow:Restoring parameters from /tmp/_retrain_checkpoint
INFO:tensorflow:Final test accuracy = 74.0% (N=696)
...
then from using label_image.py:

it gets 35% is correct

Have I written custom code: no
OS Platform and Distribution: macosx latest
TensorFlow installed from: pip
TensorFlow version: 1.7
Bazel version: Build label: 0.11.1-homebrew
   Build target: bazel-out/darwin- 
   opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
   Build time: Sun May 31 16:57:39 +50150 (1520426998659)
   Build timestamp: 1520426998659
   Build timestamp as int: 1520426998659

CUDA/cuDNN version: N/A
GPU model and memory: N/A
Exact command to reproduce: see above

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by jgberg** [comment]
​I've added it... But in fact all that information was there when I
originally posted, to the extent that it is relevant in this case.

Thanks​

On Fri, May 4, 2018 at 5:42 AM, Alfred Sorten Wolf <notifications@github.com

**Comment by renwang435** [comment]
I've been able to replicate this issue with TensorFlow 1.7 on a CentOS distribution running a 12 GB Tesla P100, CUDA version 8.0. Namely,  downloaded MobileNet V1 and V2 from their respective tfhub modules and retrained on roughly 7000 images using the retrain.py script, default parameters. Final test accuracy for both the v1 and v2 models is roughly 74%. When then going back and using label_image.py to validate on a separate set of roughly 20 images with the same properties as the original training set, the v1 model gives comparable performance (~75%), but the v2 model inexplicably drops to 33%.

**Comment by jgberg** [comment]
Any news from the front?

**Comment by jgberg** [comment]
Hi, It has been 21 days with no activity and the awaiting response label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by N2ITN** [comment]
I too was wondering if how to adapt this to mobilenet v2, or even mobilenet v2-SSD

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by achowdhery** [comment]
@jgberg Is this still an open issue?

**Comment by jgberg** [comment]
yes

**Comment by tensorflowbutler** [comment]
Nagging Assignees @petewarden, @achowdhery, @tatatodd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @petewarden, @achowdhery, @tatatodd: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by renwang435** [comment]
Can confirm this is still an issue. Even validating on the very set of images the stock MobileNetV2 model is trained on leads to inexplicably lower accuracy. Is the problem in the validation scripts provided or the way the models are loaded?

**Comment by tensorflowbutler** [comment]
Nagging Assignees @petewarden, @achowdhery, @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @petewarden, @achowdhery, @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by damhurmuller** [comment]
Same problem here

**Comment by tensorflowbutler** [comment]
Nagging Assignees @petewarden, @achowdhery, @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @petewarden, @achowdhery, @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by petewarden** [comment]
My guess is that there's something wrong with the input image preprocessing in this case, in particular the input_mean and input_std values (though if it's working for v1, it should be similar for v2). Since this is now a bit old, closing, but please reopen with details on the exact command lines if you're still hitting the problem.
